code/clue.js

```diff
@@ -7,53 +7,267 @@ const mrGreen = {
+  color: 'purple',
+  description: 'A grand dame who maintains her dignity',
+  age: 32,
+  image: 'assets/peacock.png',
+  occupation: 'Socialite',
+  favouriteWeapon: 'candlestick'
+};
+
+const colMustard = {
+  firstName: 'Colonel Jack',
+  lastName: 'Mustard',
+  color: 'yellow',
+  description: 'A dapper colonial imperalist and dangerous military man',
+  age: 55,
+  image: 'assets/mustard.png',
+  occupation: 'Colonel',
+  favouriteWeapon: 'dumbbell'
+};
+
+const mrsWhite = {
+  firstName: 'Mrs',
+  lastName: 'White',
+  color: 'white',
+  description: 'A two-faced servant who smiles politely but sneers behind the back',
+  age: 60,
+  image: 'assets/white.png',
+  occupation: 'Housekeeper',
+  favouriteWeapon: 'knife'
+};
+
+// CREATE OBJECTS FOR ALL THE WEAPONS:
 
 const rope = {
-  name: 'rope',
-  weight: 10
-}
+  name: 'Rope',
+  weight: 5,
+  damage: 2,
+  image: 'assets/rope.png'
+};
+
+const knife = {
+  name: 'Knife',
+  weight: 3,
+  damage: 9,
+  image: 'assets/knife.png'
+};
+
+const candlestick = {
+  name: 'Candlestick',
+  weight: 8,
+  damage: 5,
+  image: 'assets/candlestick.png'
+};
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const dumbbell = {
+  name: 'Dumbbell',
+  weight: 10,
+  damage: 8,
+  image: 'assets/dumbbell.png'
+};
+
+const poison = {
+  name: 'Poison',
+  weight: 1,
+  damage: 10,
+  image: 'assets/poison.png'
+};
+
+const axe = {
+  name: 'Axe',
+  weight: 7,
+  damage: 9,
+  image: 'assets/axe.png'
+};
+
+const bat = {
+  name: 'Bat',
+  weight: 5,
+  damage: 8,
+  image: 'assets/bat.png'
+};
+
+const trophy = {
+  name: 'Trophy',
+  weight: 7,
+  damage: 5,
+  image: 'assets/trophy.png'
+};
+
+const pistol = {
+  name: 'Pistol',
+  weight: 2,
+  damage: 9,
+  image: 'assets/pistol.png'
+};
 
 // NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
 
 const suspects = [
   mrGreen,
+  profPlum,
+  msScarlet,
+  mrsPeacock,
+  colMustard,
   mrsWhite
-  // ...  and the rest
-]
+];
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+];
 
-const rooms = []
+const rooms = [
+  'Dining Room',
+  'Conservatory',
+  'Kitchen',
+  'Study',
+  'Library',
+  'Billiard Room',
+  'Lounge',
+  'Ballroom',
+  'Hall',
+  'Spa',
+  'Living Room',
+  'Observatory',
+  'Theater',
+  'Guest House',
+  'Patio'
+];
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
+// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION:
 const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+  return array[Math.floor(Math.random() * array.length)];
+};
+
+// OBJECT THAT KEEPS THE MYSTERY:
+const mystery = {
+  // The values will be set later after running the pickKiller, pickWeapon and pickRoom functions
+};
```

Comment:
Defining the initial values would give the reader an idea of what kind of values to expect here later.  That is useful.  By defining `mystery.weapon`, `mystery.killer`, and `mystery.room`, you are implying not to expect something like `mystery.accomplice` or `mystery.vehicle`